components/magic.js

```diff
@@ -0,0 +1,32 @@
+async function cardCmd(message, args)
+{
+    searchCard(args[0])
+    .then((card) => {message.channel.send(card.image);})
+}
+
+function searchCard(searchTerm) {
+    return new Promise((resolve, reject) =>
+    {
+        let searchLink = "https://api.scryfall.com/cards/named?fuzzy=" + searchTerm;
+        fetch(searchLink)
+        .then(response => {
+            if(!response.ok) {
+                reject("[Scryfall API failed]");
+            }
+            return response.json();
+        })
+        .then(data => {
```

Comment:
Why do you need a second .then() here? Can't you take the response.json() above and format it  and resolve it right there?

Reply:
yeah I thought that but I copied off of my bad code from 3 years ago and I trusted my past self too much